common/app/components/Player.tsx

```diff
@@ -241,17 +255,27 @@ export default function Player({
         pause(clock, mediaAdapter, true);
     }, [playMode, clock, mediaAdapter, videoFileUrl, settings]);
 
+    const getCurrentSubtitle = (currentTime: number) => {
+        const subtitlesAtCurrentTime = subtitleCollection.subtitlesAt(currentTime);
+        return subtitlesAtCurrentTime.showing.length > 0 ? subtitlesAtCurrentTime.showing[0] : null;
+    };
+
     const handleOnWillStopShowingSubtitle = useCallback(() => {
-        if (
-            playMode !== PlayMode.autoPause ||
-            settings.autoPausePreference !== AutoPausePreference.atEnd ||
-            videoFileUrl // Let VideoPlayer do the auto-pausing
+        const currentTime = clock.time(calculateLength());
+        const currentSubtitle = getCurrentSubtitle(currentTime);
```

Comment:
The `currentSubtitle` calculation here is not necessary (sorry it sort of triggers my OCD that this calculation is being done again). The subtitle can be passed through from `AutoPauseContext` if you modify the signature of the `willStopShowingSubtitle` hook.

Reply:
Changed